cs3api4lab/cs3_file_api.py

```diff
@@ -0,0 +1,278 @@
+"""
+cs3_file_api.py
+
+CS3 File API for the JupyterLab Extension
+
+Authors:
+"""
+
+import http
+import time
+
+import cs3.gateway.v1beta1.gateway_api_pb2 as cs3gw
+import cs3.gateway.v1beta1.gateway_api_pb2_grpc as cs3gw_grpc
+import cs3.rpc.code_pb2 as cs3code
+import cs3.storage.provider.v1beta1.provider_api_pb2 as cs3sp
+import cs3.storage.provider.v1beta1.resources_pb2 as cs3spr
+import grpc
+import requests
+
+
+class Cs3FileApi:
+	ctx = {}  # "map" to store some module context: cf. init()
+	tokens = {}  # map userid [string] to {authentication token, token expiration time}
+
+	def __init__(self, config, log):
+		"""
+		Init module-level variables
+		"""
+		self.ctx['log'] = log
+		self.ctx['chunksize'] = int(config['chunksize'])
+		self.ctx['authtokenvalidity'] = int(config['authtokenvalidity'])
+
+		reva_host = config['revahost']
+
+		# prepare the gRPC connection
+		ch = grpc.insecure_channel(reva_host)
+		self.ctx['cs3stub'] = cs3gw_grpc.GatewayAPIStub(ch)
+		return
+
+	def __authenticate(self, userid):
```

Comment:
It's impossible to change because method names are imported from cs3apis package

Reply:
I'm changing all local variable names to "snake_case"

Reply:
I'm not saying to change the methods of the objects that come from the cs3api package, those ofc need to be kept (they don't use snake case because their code is belongs to a project with other languages and is, therefore, not following completely the Python style). I'm just saying to change ours.